sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -2264,17 +2258,20 @@ object QueryCompilationErrors {
   }
 
   def usingUntypedScalaUDFError(): Throwable = {
-    new AnalysisException("You're using untyped Scala UDF, which does not have the input type " +
-      "information. Spark may blindly pass null to the Scala closure with primitive-type " +
-      "argument, and the closure will see the default value of the Java type for the null " +
-      "argument, e.g. `udf((x: Int) => x, IntegerType)`, the result is 0 for null input. " +
-      "To get rid of this error, you could:\n" +
-      "1. use typed Scala UDF APIs(without return type parameter), e.g. `udf((x: Int) => x)`\n" +
-      "2. use Java UDF APIs, e.g. `udf(new UDF1[String, Integer] { " +
-      "override def call(s: String): Integer = s.length() }, IntegerType)`, " +
-      "if input types are all non primitive\n" +
-      s"3. set ${SQLConf.LEGACY_ALLOW_UNTYPED_SCALA_UDF.key} to true and " +
-      s"use this API with caution")
+    new AnalysisException(
+      errorClass = "INVALID_UDF",
+      messageParameters = Array(
+        "You're using untyped Scala UDF, which does not have the input type " +
```

Comment:
We shouldn't pass so much text as parameters. Could you imagine that we are going to translate error messages in error-classes.json to other languages like Chinese. What can be translated, only the word "invalid" in "Invalid UDF: %s"?

Reply:
Thanks @MaxGekk. If that is the case, maybe we need to have more specific errorClass. I will make the change.

Reply:
Yep, please make more specific error classes and pass minimum needed info via parameters.